cpp/src/arrow/compute/api_scalar.h

```diff
@@ -345,6 +345,18 @@ Result<Datum> IsValid(const Datum& values, ExecContext* ctx = NULLPTR);
 ARROW_EXPORT
 Result<Datum> IsNull(const Datum& values, ExecContext* ctx = NULLPTR);
 
+/// \brief IsNan returns true for each element of `values` that is NaN,
+/// false otherwise
+///
+/// \param[in] values input to look for NaN
+/// \param[in] ctx the function execution context, optional
+/// \return the resulting datum
+///
+/// \since X.X.X
```

Comment:
This would be 3.0.0 if this PR is merged in a week or two.

Reply:
Thanks @pitrou for your review, I will update the PR today answering all points. Should I rebase first ?

Reply:
Yes, you can rebase (before or after the changes, as you prefer).